src/python/zquantum/core/interfaces/backend_test.py

```diff
@@ -0,0 +1,160 @@
+import unittest
+import numpy as np
+from pyquil import Program
+from pyquil.gates import X, CNOT, H
+from pyquil.wavefunction import Wavefunction
+from openfermion import QubitOperator, IsingOperator
+
+from ..circuit import Circuit
+from ..measurement import Measurements, ExpectationValues
+from ..bitstring_distribution import BitstringDistribution
+
+class QuantumBackendTests(object):
+    # To run tests with this base class, the following variables need to be properly initialized in the child class:
+        # self.backends
+
+    def test_run_circuit_and_measure_correct_indexing(self):
+        # Note: this test may fail with noisy devices
+        # Given
+        circuit = Circuit(Program(X(0), X(0), X(1), X(1), X(2)))
+        n_samples = 100
+        # When
+        for backend in self.backends:
+            backend.n_samples = n_samples
+            measurements = backend.run_circuit_and_measure(circuit)
+
+            # Then (since SPAM error could result in unexpected bitstrings, we make sure the most common bitstring is
+            #   the one we expect)
+            counts = measurements.get_counts()
+            self.assertEqual(max(counts, key=counts.get), "001")
+
```

Comment:
There should be just one line of break between two methods of the same class ;)
https://www.python.org/dev/peps/pep-0008/#blank-lines

Reply:
Elsewhere as well.

Reply:
Addressed here:
7dcdc1e